test/system/build-testimage

```diff
@@ -12,6 +12,76 @@
 # still need a fedora image for that.
 #
 
+if [ "$(uname -s)" = "Linux" ]; then
+    testimg_base=quay.io/libpod/testimage
+    arch_list="amd64 arm64 ppc64le s390x"
+
+    function busyboximg {
+        local repo=$1
+        echo docker.io/$repo/busybox:1.34.1
+    }
+
+    function build_containerfile {
+        # alpine because it's small and light and reliable
+        #    - check for updates @ https://hub.docker.com/_/alpine
+        # busybox-extras provides httpd needed in 500-networking.bats
+        # iproute2 provides JSON output (not in busybox) for 505-networking-pasta.bats
+        # socat offers convenient UDP test termination in 505-networking-pasta.bats
+        cat <<EOF
+ARG REPO=please-override-repo
+FROM docker.io/\${REPO}/alpine:3.16.2
+RUN apk add busybox-extras iproute2 socat
+ADD testimage-id pause /home/podman/
+RUN rm -f /var/cache/apk/*
+EOF
+    }
+
+    # docker.io repo is usually the same name as the desired arch; except
+    # for arm64, where podman needs to have the arch be 'arm64' but the
+    # image lives in 'arm64v8'.
+    function repo_for_arch {
+        local arch=$1
+        if [[ $repo = "arm64" ]]; then
+            echo ${repo}v8
+        else
+            echo ${repo}
+        fi
+    }
+elif [ "$(uname -s)" = "FreeBSD" ]; then
+    testimg_base=quay.io/dougrabson/testimage
+    arch_list="amd64 arm64"
+    busybox=quay.io/dougrabson/busybox:1.36.1
+
+    function date {
+        gdate "$@"
+    }
+    function mktemp {
+        gmktemp "$@"
+    }
+    function touch {
+        gtouch "$@"
+    }
+
+    function busyboximg {
+        echo quay.io/dougrabson/busybox:1.36.1
+    }
+
+    function build_containerfile {
+        # busybox because it's small and has a command set which is close to what
+        # the test suite needs.
+        cat <<EOF
+FROM ${busybox}
```

Comment:
This is the crux; this is why I feel obligated to block this. This and line 51, but mostly this.

The source image we use for `testimage`, `docker.io/ARCH/alpine`, is not under our control but is presumably a known quantity. This one is a black box. It is presumably a busybox image that you have crafted yourself, either from scratch (my guess) or based on an existing container image from somewhere else. It's basically magic.

Which is OK for an image used only by you, ... but this is not something that we (upstream) can in any way test, verify, maintain, or support. This is basically asking upstream to carry a very large complicated set of patches for something that upstream can't possibly work with. 

I'm really sorry. I can't approve this.